app/views/compute_resources_vms/form/vmware/_network.html.erb

```diff
@@ -2,11 +2,7 @@
   :class => "col-md-3 vmware_type",
   :label => _('NIC type'), :label_size => "col-md-3", :disabled => !new_vm
 %>
-<% if new_vm %>
-  <%= select_f f, :network, vsphere_networks(compute_resource), :first, :last, { },
-    :class => "col-md-3 vmware_network",
-    :label => _('Network'), :label_size => "col-md-3"
-  %>
-<% else %>
-  <%= text_f f, :network, :class => 'col-md-3 vmware_network', :label => _("Network"), :label_size => "col-md-3", :readonly => true %>
```

Comment:
This was an input field intentionally and was chosen to speed up page loading. Can you find a solution that does not load the complete list of networks from vSphere? I believe we have improved the API calls in fog-vsphere recently, but we should do our best to keep this as fast as possible.

Reply:
I have investigated it a bit and unfortunately it is not possible.

This is closest I could get to what you ask - but the performance is better just on frontend side:
https://github.com/ezr-ondrej/foreman/commit/05393d067898d5db037b183a4cffd44ed6e59f79

Unfortunately, the fog-vsphere `networks.get(id)` is not working at the moment.
I will work on fix for that, than it could make sense to opt the code a bit and make it work.
For now I don't see it really relevant, but I can clean it up a bit and add it to this PR.
It would prepare the codebase for the fog-vsphere fix. Or we can actually monkey patch the implementation till that time :)

Reply:
Ok, that's even worse. I'm actually ok with this implementation. Let me test it and we're good to go.